CMakeLists.txt

```diff
@@ -84,87 +80,68 @@ target_link_libraries(rmm INTERFACE spdlog::spdlog_header_only)
 target_link_libraries(rmm INTERFACE dl)
 target_compile_features(rmm INTERFACE cxx_std_17 $<BUILD_INTERFACE:cuda_std_17>)
 
-# Set logging level. Must go before including gtests and benchmarks.
-
-set(RMM_LOGGING_LEVEL
-    "INFO"
-    CACHE STRING "Choose the logging level.")
-# Set the possible values of build type for cmake-gui
-set_property(CACHE RMM_LOGGING_LEVEL PROPERTY STRINGS "TRACE" "DEBUG" "INFO" "WARN" "ERROR"
-                                              "CRITICAL" "OFF")
-message(STATUS "RMM: RMM_LOGGING_LEVEL = '${RMM_LOGGING_LEVEL}'")
-
 if((BUILD_TESTS OR BUILD_BENCHMARKS) AND CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
-  # Auto-detect available GPU compute architectures
-  include(${RMM_SOURCE_DIR}/cmake/Modules/SetGPUArchs.cmake)
-  # Enable the CUDA language after setting CMAKE_CUDA_ARCHITECTURES
+  include(rapids-cuda)
+  rapids_cuda_init_architectures(RMM)
   enable_language(CUDA)
+
+  # Since RMM only enables CUDA optionally we need to manually include the file that
+  # rapids_cuda_init_architectures relies on `project` calling
+  include("${CMAKE_PROJECT_RMM_INCLUDE}")
   message(STATUS "RMM: Building benchmarks with GPU Architectures: ${CMAKE_CUDA_ARCHITECTURES}")
 endif()
 
 # optionally build tests
-
 if(BUILD_TESTS AND CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
-  enable_testing()
+  include(cmake/thirdparty/get_gtest.cmake)
```

Comment:
Why not do this include inside the tests/CMakeLists.txt? Same question for the benchmarks one.

Reply:
Honestly it was a style choice. I like the appearance of having all the `include()` inside a single CMake file.
So projects like cugraph ( https://github.com/rapidsai/cugraph/blob/branch-21.08/cpp/CMakeLists.txt#L138 ) which have a consolidated list of dependencies makes me happy. 

Now `rmm` isn't my project, so I can switch it back to having the gtest and gbench includes occur inside the relevant CMakeLists. Up to you